R/sce_to_seurat.R

```diff
@@ -0,0 +1,82 @@
+
+#' Convert SingleCellExperiment object to Seurat object
+#'
+#' @param sce SingleCellExperiment object
+#'
+#' @return Seurat object
+#'
+#' @import SingleCellExperiment
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' sce_to_seurat(sce = sce_object)
+#' }
+sce_to_seurat <- function(sce){
+
+  if (!requireNamespace("Seurat", quietly = TRUE)) {
+    stop("The Seurat package must be installed to create a Seurat object. No output returned.")
+  }
+
+  if(!is(sce,"SingleCellExperiment")){
+    stop("Input must be a SingleCellExperiment object.")
+  }
+
+  # Seurat counts need to be integers
+  sce_counts <- round(counts(sce))
+  # Seurat will not like zero count cells
+  sce_sum <- Matrix::colSums(sce_counts)
+  seurat_cells <- names(sce_sum)[sce_sum > 0]
+  sce_counts <- sce_counts[, seurat_cells]
+
+  # convert metadata to Seurat compatible formats
+  coldata <- as.data.frame(colData(sce))
+  rowdata <- as.data.frame(rowData(sce))
+
+
+  # create seurat object
+  seurat_obj <- Seurat::CreateSeuratObject(counts = sce_counts,
+                                           meta.data = coldata)
+
+  # add rowdata and metadata separately adding it while creating leaves the slots empty without warning
+  seurat_obj[["RNA"]]@var.features <- rowdata
+  seurat_obj@misc <- metadata(sce)
```

Comment:
Just wondering if we want to remove the model here. I don't think we would need it in any Seurat application.